File: frontend/src/types/bet.ts
export type ResultadoAposta = "pendente" | "ganha" | "perdida" | "anulada" | "cashout";

export interface BetFixtureSummary {
  id: number;
  time_casa: string;
  time_fora: string;
  data_hora: string;
}

export interface Bet {
  id: number;
  fixture_id: number;
  analysis_id: number | null;
  mercado: string;
  selecao: string;
  odd: number;
  stake: number;
  resultado: ResultadoAposta;
  lucro: number | null;
  created_at: string;
  fixture: BetFixtureSummary | null;
}

export interface BetCreate {
  fixture_id: number;
  mercado: string;
  selecao: string;
  odd: number;
  stake: number;
  analysis_id?: number | null;
}

export interface BetUpdate {
  resultado: ResultadoAposta;
  lucro?: number | null;
}

export interface BankrollPoint {
  data: string;
  saldo_acumulado: number;
}

export interface BetStats {
  total_apostas: number;
  apostas_resolvidas: number;
  taxa_acerto: number | null;
  total_investido: number;
  lucro_acumulado: number;
  roi: number | null;
  yield_: number | null;
  evolucao: BankrollPoint[];
}

File: backend/app/schemas/bet.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.bet import ResultadoAposta


class BetFixtureSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    time_casa: str
    time_fora: str
    data_hora: datetime


class BetCreate(BaseModel):
    fixture_id: int
    mercado: str
    selecao: str
    odd: float
    stake: float
    analysis_id: int | None = None


class BetUpdate(BaseModel):
    resultado: ResultadoAposta
    lucro: float | None = None


class BetOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    fixture_id: int
    analysis_id: int | None
    mercado: str
    selecao: str
    odd: float
    stake: float
    resultado: ResultadoAposta
    lucro: float | None
    created_at: datetime
    fixture: BetFixtureSummary | None = None


class BetStatsOut(BaseModel):
    total_apostas: int
    apostas_resolvidas: int
    taxa_acerto: float | None
    total_investido: float
    lucro_acumulado: float
    roi: float | None
    yield_: float | None
    evolucao: list["BankrollPoint"]


class BankrollPoint(BaseModel):
    data: datetime
    saldo_acumulado: float


BetStatsOut.model_rebuild()
